intermediate_oop/tic_tac_toe/js/board.js

```diff
@@ -0,0 +1,107 @@
+// board
+
+function Board() {
+	this.player = Math.random() < 0.5 ? "X" : "O"; // randomize starting player
+	this.turnCount = 0;
+	this.squares = {}; // keys are squareIds, values are the square instances
+	this.gameBoard = [[],[],[]]; // nested array to store played moves
+	this.winner = "none";
+}
+
+// function to add squares to the board
+Board.prototype.addSquares = function (squareIds) {
```

Comment:
rather than calling this function separately, it may make sense to take an array of square Ids in the constructor function